blocks/basic/include/gnuradio-4.0/basic/clock_source.hpp

```diff
@@ -130,32 +125,41 @@ The 'tag_times[ns]:tag_value(string)' vectors control the emission of tags with
             }
         }
 
-        auto samplesToNextTag = tags.empty() || _nextTag >= tags.size() ? std::numeric_limits<uint32_t>::max() : static_cast<gr::Size_t>(tags[_nextTag].index) - n_samples_produced;
+        auto samplesToNextTag = tags.empty() || _nextTagIndex >= tags.size() ? std::numeric_limits<uint32_t>::max() : static_cast<gr::Size_t>(tags[_nextTagIndex].index) - n_samples_produced;
 
         if (samplesToNextTag < samplesToNextTimeTag) {
-            if (_nextTag < tags.size() && samplesToNextTag <= samplesToProduce) {
-                const auto signedSamplesProduced = static_cast<Tag::signed_index_type>(n_samples_produced);
-                const auto tagDeltaIndex         = tags[_nextTag].index - signedSamplesProduced; // position w.r.t. start of this chunk
+            if (_nextTagIndex < tags.size() && samplesToNextTag <= samplesToProduce) {
+                const auto tagDeltaIndex = tags[_nextTagIndex].index - static_cast<Tag::signed_index_type>(n_samples_produced); // position w.r.t. start of this chunk
                 if (verbose_console) {
-                    gr::testing::print_tag(tags[_nextTag], fmt::format("{}::processBulk(...)\t publish tag at  {:6}", this->name, signedSamplesProduced + tagDeltaIndex));
+                    gr::testing::print_tag(tags[_nextTagIndex], fmt::format("{}::processBulk(...)\t publish tag at  {:6}", this->name, n_samples_produced + tagDeltaIndex));
                 }
-                out.publishTag(tags[_nextTag].map, tagDeltaIndex);
+                out.publishTag(tags[_nextTagIndex].map, tagDeltaIndex);
                 samplesToProduce = samplesToNextTag;
-                _nextTag++;
+                _nextTagIndex++;
             }
         } else {
             if (!tag_times.value.empty() && _nextTimeTag < tag_times.value.size() && samplesToNextTimeTag <= samplesToProduce) {
-                std::string  value    = _nextTimeTag < tag_values.value.size() ? tag_values.value[_nextTimeTag] : do_zero_order_hold ? tag_values.value.back() : "";
-                property_map context  = { { "context", value } };
-                property_map metaInfo = { { "trigger_meta_info", context } };
+                const std::string     value = _nextTimeTag < tag_values.value.size() ? tag_values.value[_nextTimeTag] : do_zero_order_hold ? tag_values.value.back() : "";
+                std::string           triggerName;
+                [[maybe_unused]] bool triggerNameNegated;
+                std::string           triggerContext;
+                [[maybe_unused]] bool triggerContextNegated;
+                gr::basic::trigger::detail::parse(value, triggerName, triggerNameNegated, triggerContext, triggerContextNegated);
+                property_map triggerTag;
```

Comment:
Agree. We should rediscuss the invariants and what a Tag should contain by default. This also for performance reasons because the pmtv::map_t isn't as cheap as it could be for the case where we have 30k trigger/tags per seconds or more....